src/rest.rs

```diff
@@ -36,29 +34,16 @@ impl Client {
         Ok(Client {
             config,
             address,
-            token: token,
+            token,
             rest: client,
         })
     }
 
-    pub async fn req<T: Serialize + ?Sized>(
-        &self,
-        url: String,
-        method: Method,
-        body: Option<&T>,
-    ) -> Result<Response, BrigadeError> {
-        let token = self.token.clone().unwrap();
-        let req = self.rest.request(method, &url).bearer_auth(token);
-        let res = match body {
-            Some(body) => req.json(body).send().await?,
-            None => req.send().await?,
-        };
-
-        Ok(res)
+    pub fn req<U: IntoUrl>(&self, method: Method, url: U) -> RequestBuilder {
+        let req = self.rest.request(method, url);
+        match self.token.clone() {
```

Comment:
Nit: to avoid an unneeded clone, you could do `self.token.as_ref()` and then do `t.clone()` only when you need it